File: gitdot-api/src/endpoint/organization/upload_organization_image.rs
use crate::endpoint::Endpoint;

pub struct UploadOrganizationImage;

impl Endpoint for UploadOrganizationImage {
    const PATH: &'static str = "/organization/{org_name}/image";
    const METHOD: http::Method = http::Method::POST;

    type Request = ();
    type Response = ();
}

File: gitdot-web/packages/gitdot-api/src/endpoint/organization/upload-organization-image.ts
import { z } from "zod";
import type { Endpoint } from "../endpoint";

export const UploadOrganizationImage = {
  path: "/organization/{org_name}/image",
  method: "POST",
  request: z.instanceof(Blob),
  response: z.void(),
} as const satisfies Endpoint;
export type UploadOrganizationImage = typeof UploadOrganizationImage;
